src/interface/ArgInterface.py

```diff
@@ -23,14 +46,98 @@ def _service_date(self, arg):
             # logger.log(err_msg, Severity.ERROR)
             raise argparse.ArgumentTypeError(err_msg)
 
-    def _parse_cl_args(self, args):
+    def _service_year(self, arg):
+        try:
+            return datetime.strptime(arg, "%Y")
+        except ValueError:
+            err_msg = "Invalid service year format: {0}, YYYY expected.".format(arg)
+            # TODO uncomment when ready to enable logging output
+            # logger.log(err_msg, Severity.ERROR)
+            raise argparse.ArgumentTypeError(err_msg)
+
+    def _service_period(self, arg):
+        try:
+            try:
+                p = int(arg)
+                if p in [1, 2, 3]:
+                    return p
+                else:
+                    raise ValueError
+            except ValueError as e:
+                names = ["first", "second", "third"]
+                if arg in names:
+                    return names.index(arg) + 1
+                else:
+                    raise e
+        except ValueError:
+            err_msg = "Invalid service period format: {0}, [1, 2, 3] or [first, second, third] expected.".format(arg)
+            # TODO uncomment when ready to enable logging output
+            # logger.log(err_msg, Severity.ERROR)
+            raise argparse.ArgumentTypeError(err_msg)
+
+    def _limit(self, arg):
+        try:
+            lim = int(arg)
+            if lim > 0:
+                return lim
+            else:
+                raise ValueError
+        except ValueError:
+            err_msg = "Invalid limit: {0}, value must be at least 1.".format(arg)
+            # TODO uncomment when ready to enable logging output
+            # logger.log(err_msg, Severity.ERROR)
+            raise argparse.ArgumentTypeError(err_msg)
+
+    def _create_parser(self, args):
+        if args is None:
+            raise SystemExit(2)
```

Comment:
Why do you raise a system exit in several places instead of returning none or doing something less extreme?

Reply:
Characteristic of ArgParse is to raise a SystemExit if arguments are improper, due to the fact that it reads from initialization parameters, and there will never be a "happy path" out of an error. I mimic this behavior under similar circumstances for consistency. I intend to expand on the logger functionality when we move to fully integrate it so that the errors are made more readily available for viewing.